lib/ASRouter.jsm

```diff
@@ -1777,6 +1780,44 @@ class _ASRouter {
     this.onMessage({ data: action, target });
   }
 
+  async loadNewTabMessages(options = {}, target) {
+    const { endpoint, url } = options;
+    let message;
+
+    // Load preview endpoint for snippets if one is sent
+    if (endpoint) {
+      await this._addPreviewEndpoint(endpoint.url, target.portID);
+    }
+
+    // Load all messages
+    await this.loadMessagesFromAllProviders();
+
+    // On about welcome, set up trailhead experiments
+    if (url === "about:welcome") {
+      Services.prefs.setBoolPref(
+        TRAILHEAD_CONFIG.DID_SEE_ABOUT_WELCOME_PREF,
+        true
+      );
+      await this.setupTrailhead();
+
+      message = await this.handleMessageRequest({
+        provider: "onboarding",
+        triggerId: "firstRun",
+      });
+    } else {
+      // On new tab, send cards if they match; othwerise send a snippet
+      message =
+        (await this.handleMessageRequest({
```

Comment:
handleMessageRequest expects a `triggerId` otherwise it will filter out `EXTENDED_TRIPLETS_1` because the message trigger id is compared to `undefined` (no cards are loaded on a regular new tab).